spec/helpers/cd4pe_job_helper_spec.rb

```diff
@@ -0,0 +1,194 @@
+require 'open3'
+require_relative '../../lib/puppet_x/puppetlabs/cd4pe_job_helper.rb'
+
+describe 'cd4pe_job_helper' do
+  before(:each) do
+    @working_dir = File.join(Dir.getwd, "test_working_dir")
+    Dir.mkdir(@working_dir)
+  end
+
+  after(:each) do
+    FileUtils.remove_dir(@working_dir)
+    $stdout = STDOUT
+  end
+
+  describe 'cd4pe_job_helper::initialize' do
+    it 'Converts the docker run args into a usable string.' do
+      arg1 = '--testarg=woot'
+      arg2 = '--otherarg=hello'
+      arg3 = '--whatever=yolo'
+      user_specified_docker_run_args = [arg1, arg2, arg3]
+  
+      job_helper = PuppetX::Puppetlabs::CD4PEJobHelper.new(working_dir: @working_dir, docker_run_args: user_specified_docker_run_args)
+  
+      expect(job_helper.docker_run_args).to eq("#{arg1} #{arg2} #{arg3}")
+    end
+  end
+
+  describe 'cd4pe_job_helper::set_job_env_vars' do
+    it 'Sets the job environment params.' do
+      job_helper = PuppetX::Puppetlabs::CD4PEJobHelper.new(working_dir: @working_dir)
+  
+      cd4pe_web_ui_endpoint = 'https://testtest.com'
+      cd4pe_token = 'alksjdbhfnadhsbf'
+      cd4pe_job_owner = 'carls cool carl'
+      job_instance_id = '17'
+  
+      params = {
+        'cd4pe_web_ui_endpoint' => cd4pe_web_ui_endpoint,
+        'cd4pe_token' => cd4pe_token,
+        'cd4pe_job_owner' => cd4pe_job_owner,
+        'job_instance_id' => job_instance_id,
+      }
+  
+      job_helper.set_job_env_vars(params)
+      
+      expect(ENV['WEB_UI_ENDPOINT']).to eq(cd4pe_web_ui_endpoint)
+      expect(ENV['JOB_TOKEN']).to eq(cd4pe_token)
+      expect(ENV['JOB_OWNER']).to eq(cd4pe_job_owner)
+      expect(ENV['JOB_INSTANCE_ID']).to eq(job_instance_id)
+    end
+  
+    it 'Sets the user-specified environment params.' do
+      job_helper = PuppetX::Puppetlabs::CD4PEJobHelper.new(working_dir: @working_dir)
+  
+      user_specified_env_vars = [
```

Comment:
see earlier comment re: expectations around well formed `env_vars`

Reply:
We don't care in this context, as the caller is responsible for ensuring the integrity of these values.

In our case, the caller is CD4PE, and I believe we DO indeed verify the integrity of these values when we save them to a job.